spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -213,13 +214,26 @@ public Scan build() {
         SparkReadOptions.END_SNAPSHOT_ID,
         SparkReadOptions.START_SNAPSHOT_ID);
 
+    checkInvalidOptions();
+
     if (startSnapshotId != null) {
       return buildIncrementalAppendScan(startSnapshotId, endSnapshotId);
     } else {
       return buildBatchScan(snapshotId, asOfTimestamp, branch, tag);
     }
   }
 
+  private void checkInvalidOptions() {
+    Long startTimestamp = readConf.startTimestamp();
```

Comment:
Nit: maybe just inline this if its only used in one place (unless I miss something)?  Doesnt make sense to make a helper method called 'checkInvalidOption' but only add one specific check.

Reply:
I thought it'd be more readable. No? I'm OK with either.

Reply:
Inlined in the new commit.